### PR TITLE
Rwahs 52/item data

### PR DESCRIPTION
**Pull request by kehh**
- runAllImports now populates the search index afterwars rather than on the fly
- Add sample data for `itemData`
- Does not include Subject List which is now in https://youtrack.gaiaresources.com.au/issue/RWAHS-69
- ObjectExport view updated to synthesise new fields and also return `null` for all empty fields.
- Corresponds to the `itemData` filter view in the import mapping spreadsheet

### CODE REVIEW

**Comment by kehh** [comment]
Requires https://github.com/rwahs/installation-profile/pull/5

File: museum/sampleRecords/itemData.destination.json [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+      "id":"2",
+      "idno":"A1998.16",
+      "display_label":"GALLOP HOUSE",
+      "ca_objects.Description":"Harper, Mary Alison. Unframed, unglazed oil painting of Gallop House overlooking Swan River. Signed: LRC.  Painted c1974.",
+      "ca_objects.MakersMarks":"Harper, Mary Alison",
+      "ca_objects.ItemDates":"c1974",
+      "ca_objects.PhysicalCondition":"2",
+      "ca_objects.Materials":"Oil on Masonite",
+      "ca_objects.Dimensions.DimensionsValue":"H: 45.4cms W: 55.5cms.  Unframed.",
+      "classificationHierarchy":"ILLUSTRATIONS;Oil;building"
```
Comment (leftclickben) [inline museum/sampleRecords/itemData.destination.json]: This `classificationHierarchy` value does not include the root, does this mean "ILLUSTRATIONS" is making a new root?

File: museum/sampleRecords/itemData.destination.json [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+      "ca_objects.Description":"Brown wood with gilt line frame, glazed, engraving\/ etching of Perth Town Hall clock tower &amp; people on Hay Street corner, painted c1916. Signed LL.",
+      "ca_objects.MakersMarks":"Van Raalte, Henri Benedictus (HenrI)",
+      "ca_objects.ItemDates":"1916",
+      "ca_objects.EarliestYear":"1916",
+      "ca_objects.LatestYear":"1916",
+      "ca_objects.HistoricalDetails":"Henri Van Raalte 1881-1929.   Refer notes D. Erickson's Audit 2014",
+      "ca_objects.PhysicalCondition":"2",
+      "ca_objects.Materials":"Etching on paper",
+      "ca_objects.Dimensions.DimensionsValue":"H: 39.3cms W: 21cms. Frame:",
+      "classificationHierarchy":"Root node for Classification;ILLUSTRATIONS;Etching;building"
```
Comment (leftclickben) [inline museum/sampleRecords/itemData.destination.json]: (from above)... compared to this one which does include the root node.

File: museum/sampleRecords/itemData.destination.json [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+      "display_label":"COLONIAL HOSPITAL, ST MARY'S CATHEDRAL &amp; R.C. BISHOPS HOUSE",
+      "ca_objects.Description":"Colonial Hospital, St Mary's Cathedral &amp; Roman Catholic Bishops House. Victoria Square c1860s\n\nSite of Royal Perth Hospital",
+      "ca_objects.MakersMarks":"Unknown",
+      "ca_objects.ItemDates":"c1863",
+      "ca_objects.EarliestYear":"1863",
+      "ca_objects.LatestYear":"1863",
+      "ca_objects.SignificanceCriteria":"Historic;Social or Spiritual",
+      "ca_objects.StatementOfSignificance":"Historic: \nSocial:",
+      "ca_objects.Dimensions.DimensionsValue":"Large size 28x22 cm",
+      "classificationHierarchy":"Building;hospitals"
```
Comment (leftclickben) [inline museum/sampleRecords/itemData.destination.json]: This one doesn't seem to include the top-level value, which considering other values should be "PLACE"

File: museum/sampleRecords/itemData.destination.json [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+      "object_id":"7",
+      "id":"7",
+      "idno":"P1999.1013Copied",
+      "display_label":"DESCRIPTION OF 'PARK BUNGALOW' HOME OF C.Y. O'CONNOR.",
+      "ca_objects.Description":"Description of 'Park Bungalow', home of C.Y. O'Connor\nCOPIED",
+      "ca_objects.MakersMarks":"unknown",
+      "ca_objects.Importance":"State",
+      "ca_objects.StatementOfSignificance":"historical importance",
+      "ca_objects.Dimensions.DimensionsValue":"Imperial size (25x19 cm)",
+      "classificationHierarchy":"PLACE;Building;description"
```
Comment (leftclickben) [inline museum/sampleRecords/itemData.destination.json]: (from previous comment)... for example this one has "PLACE" above "Building".  Also the tertiary value "description" seems odd.

**Comment by leftclickben** [comment]
Only issues seem to be with the Classification field/hierarchy as noted.

File: museum/sampleRecords/itemData.destination.json (hunk 3 above)
Comment (kehh) [inline museum/sampleRecords/itemData.destination.json]: This appears to be a cacheing issue in the browse web service. When querying for this single record, the `classificationHierarchy` comes out correct.

``` json
{
  "ok":true,
  "results":[
    {
      "object_id":"393",
      "id":"393",
      "idno":"P1999\/1006",
      "display_label":"COLONIAL HOSPITAL, ST MARY'S CATHEDRAL &amp; R.C. BISHOPS HOUSE",
      "ca_objects.Description":"Colonial Hospital, St Mary's Cathedral &amp; Roman Catholic Bishops House. Victoria Square c1860s\n\nSite of Royal Perth Hospital",
      "ca_objects.MakersMarks":"Unknown",
      "ca_objects.ItemDates":"c1863",
      "ca_objects.EarliestYear":"1863",
      "ca_objects.LatestYear":"1863",
      "ca_objects.SignificanceCriteria":"Historic;Social or Spiritual",
      "ca_objects.StatementOfSignificance":"Historic: \nSocial:",
      "ca_objects.Dimensions.DimensionsValue":"Large size 28x22 cm",
      "classificationHierarchy":"Root node for Classification;PLACE;Building;hospitals"
    }
  ]
}
```

File: museum/sampleRecords/itemData.destination.json (hunk 4 above)
Comment (kehh) [inline museum/sampleRecords/itemData.destination.json]: That's in the source data: https://github.com/rwahs/import-scripts/pull/3/files#diff-98ed9e1e98dcdc16c390107fa6ce9e0dR110

File: museum/sampleRecords/itemData.destination.json (hunk 2 above)
Comment (kehh) [inline museum/sampleRecords/itemData.destination.json]: I'll try using a `<unit/>` tag to see if I can make this better.

Comment (kehh) [inline museum/sampleRecords/itemData.destination.json]: ok this looks like it's working now. I ended up using `<unit relativeTo='ca_list_items'><unit relativeTo='ca_list_items.hierarchy'><unit>^idno (^item_id)</unit></unit></unit`

File: museum/sampleRecords/itemData.destination.json (hunk 4 above)
Comment (leftclickben) [inline museum/sampleRecords/itemData.destination.json]: So it is.

File: museum/sampleRecords/itemData.destination.json (hunk 3 above)
Comment (leftclickben) [inline museum/sampleRecords/itemData.destination.json]: OK.  Now that I think about it, this issue seems somehow familiar.

File: museum/sampleRecords/itemData.destination.json (hunk 2 above)
Comment (leftclickben) [inline museum/sampleRecords/itemData.destination.json]: OK cool, yeah it looks good.  Having the `item_id` there as well is useful to check that there aren't duplicates being created (i.e. multiple list items with different IDs but the same label).

**Pull request merged**